sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_patch.py

```diff
@@ -27,6 +31,57 @@
 JSON = MutableMapping[str, Any]  # pylint: disable=unsubscriptable-object
 T = TypeVar("T")
 ClsType = Optional[Callable[[PipelineResponse[HttpRequest, HttpResponse], T, Dict[str, Any]], Any]]
+PollingReturnType_co = TypeVar("PollingReturnType_co", covariant=True)
+_FINISHED = frozenset(["succeeded", "canceled", "failed", "completed"])
+
+
+def _parse_operation_id(operation_location_header):
+    regex = "[^:]+://[^/]+/documentintelligence/.+/([^?/]+)"
+    return re.match(regex, operation_location_header).group(1)
+
+
+class AnalyzeDocumentLROPollingMethod(LROBasePolling):
+    @property
+    def operation_id(self) -> str:
+        """Analyze operation result ID.
+
+        :rtype:str
+        """
+        return _parse_operation_id(self._initial_response.http_response.headers["Operation-Location"])
+
+
+class AnalyzeDocumentLROPoller(LROPoller[PollingReturnType_co]):
+    def polling_method(self) -> AnalyzeDocumentLROPollingMethod:
+        """Return the polling method associated to this poller.
+
+        :return: AnalyzeDocumentLROPollingMethod
+        :rtype: AnalyzeDocumentLROPollingMethod
+        """
+        return self._polling_method  # type: ignore
+
+    @property
+    def operation_id(self) -> str:
```

Comment:
In other libraries where we have operation metadata that we need to expose, we try not to add the property directly to the poller and instead have done a `details` property which is a `Mapping[str, Any]`. This way we can easily add more operation metadata in the future

Reply:
example of where we have that https://github.com/Azure/azure-sdk-for-python/blob/5eba2a8a114eb9189731bb8779347f5c4905342d/sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_polling.py#L94

Reply:
Updated